src/types/tests/schema-test.ts

```diff
@@ -682,14 +689,39 @@ describe('schema', () => {
     const intersectT = bType.toLiteral();
     intersectT.names = [];
 
-    assert.deepEqual(Schema.intersect(aType, bType).toLiteral(), intersectT);
+    const intersectionAB = Schema.intersect(aType, bType);
+    assert.isNotNull(intersectionAB, 'intersection of A and B should exist');
+    assert.deepEqual(intersectionAB.toLiteral(), intersectT);
     assert.deepEqual(aType.toLiteral(), aTypeLit, 'input (a) to intersection should not be modified');
     assert.deepEqual(bType.toLiteral(), bTypeLit, 'input (b) to intersection should not be modified');
 
-    assert.deepEqual(Schema.intersect(bType, aType).toLiteral(), intersectT);
+    const intersectionBA = Schema.intersect(bType, aType);
+    assert.isNotNull(intersectionBA, 'intersection of B and A should exist');
+    assert.deepEqual(intersectionBA.toLiteral(), intersectT);
     assert.deepEqual(aType.toLiteral(), aTypeLit, 'input (a) to intersection should not be modified');
     assert.deepEqual(bType.toLiteral(), bTypeLit, 'input (b) to intersection should not be modified');
   };
+  const verifyIntersectOf = async (typeBuilder: (type: string) => string) => {
+    await verifyIntersectOfTypes(`${typeBuilder(`Foo {a: Text, b: Text}`)}`, `${typeBuilder(`Foo {a: Text}`)}`);
+  };
+  it('tests schema union, with nullable and non-nullable ints', async () => {
+    await verifyUnionOfTypes(`Int`, `Int?`);
+  });
```

Comment:
Not especially important, but you should be able to just pass the verify method directly:

`it('tests the thing', verifyUnionOfTypes('Int', 'Int?'));`

(That may or may not be more readable...)

Also, not sure why you're using interpolation literals; there doesn't seem to be any need for them.

Reply:
Thanks, that's neat. I'll switch the quotes too.